pkg/api/handlers/emailreceiver.go

```diff
@@ -85,16 +85,19 @@ func convertEmailReceiver(emailReceiver v1.EmailReceiver, hostname string) *type
 
 	var aliasAssigned *bool
 	if emailReceiver.Generation == emailReceiver.Status.ObservedGeneration {
-		aliasAssigned = &emailReceiver.Status.AliasAssigned
+		aliasAssigned = emailReceiver.Status.AliasAssigned
 	}
 	er := &types.EmailReceiver{
 		Metadata:              MetadataFrom(&emailReceiver),
 		EmailReceiverManifest: manifest,
-		AddressAssigned:       aliasAssigned,
+		AliasAssigned:         aliasAssigned,
 	}
 	if hostname != "" {
 		name := emailReceiver.Name
-		if er.AddressAssigned != nil && *er.AddressAssigned {
+		// We should use the alias if the following is true
+		// 1. Alias is assigned and set to true
+		// 2. Alias is not assigned but Alias is set at the initial creation.
+		if (er.AliasAssigned != nil && *er.AliasAssigned) || (er.AliasAssigned == nil && er.Alias != "") {
```

Comment:
We don't want to do it this way because if `er.AliasAssigned == nil` then we may be sending back bad data.

The way this works for agent, for example, is the UI polls to make sure it gets back a value for `AliasAssigned` (which should remain a pointer. If that value is not set, then the UI knows it should keep polling until it gets true or false.

Reply:
Ok. I think the UI only poll once when receiver is created. The problem is that the current logic returns undetermistic result from API when AliasAssigned is set, like the address will change from `name@address.com` to `alias@address.com` when controller is settled. I think we should return empty email address when er.AliasAssigned == nil and have UI poll until the alias is set.

Reply:
Yep. That sounds good to me!

Reply:
Changed the code so that when webhook/email receiver's alias is set, it will return empty address until alias is being set. Otherwise it will return the right address depends on whether alias has been set successfully or not.

The change UI will need to make is that it should start polling for alias if `alias` is set until AliasAssigned is either false or true.